File: app/src/main/java/com/VegaSolutions/lpptransit/lppapi/responseobjects/DepartureWrapper.java
package com.VegaSolutions.lpptransit.lppapi.responseobjects;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class DepartureWrapper {

    @SerializedName("station_code")
    private String stationCode;

    @SerializedName("station_name")
    private String stationName;

    @SerializedName("route_destination")
    private String routeDestination;

    @SerializedName("route_full_name")
    private String routeFullName;

    private List<Departure> departures;

    public String getStationCode() {
        return stationCode;
    }

    public String getStationName() {
        return stationName;
    }

    public String getRouteDestination() {
        return routeDestination;
    }

    public String getRouteFullName() {
        return routeFullName;
    }

    public List<Departure> getDepartures() {
        return departures;
    }

    public class Departure {

        @SerializedName("arrival_hour")
        private int arrivalHour;

        @SerializedName("arrival_minute")
        private int arrivalMinute;

        public int getArrivalHour() {
            return arrivalHour;
        }

        public int getArrivalMinute() {
            return arrivalMinute;
        }
    }

}

File: app/src/main/java/com/VegaSolutions/lpptransit/lppapi/data/DepartureWrapper.kt
package com.VegaSolutions.lpptransit.lppapi.data

import com.google.gson.annotations.SerializedName

data class DepartureWrapper(
    @SerializedName("station_code")
    val stationCode: String,

    @SerializedName("station_name")
    val stationName: String,

    @SerializedName("route_destination")
    val routeDestination: String,

    @SerializedName("route_full_name")
    val routeFullName: String,

    @SerializedName("departures")
    val departures: List<Departure>,
) {
    data class Departure (
        @SerializedName("arrival_hour")
        val arrivalHour: Int,

        @SerializedName("arrival_minute")
        val arrivalMinute: Int,
    )
}
